app/queries/rollover_progress_query.rb

```diff
@@ -44,12 +48,55 @@ def total_eligible_providers
     ).distinct
   end
 
-  def total_eligible_courses
+  def eligible_courses
     @previous_target_cycle
       .courses
       .joins(:latest_enrichment)
       .where(course_enrichment: { status: %i[published withdrawn] })
-      .where(provider_id: total_eligible_providers.select(:id)).distinct
+      .where(provider_id: eligible_providers.select(:id)).distinct
+  end
+
+  def eligible_partnerships
+    ProviderPartnership
+    .includes(:training_provider, :accredited_provider)
+    .where(
+      "accredited_provider_id IN (:ids) OR training_provider_id IN (:ids)",
+      ids: eligible_providers.select(:id),
+    )
+    .where.not(
+      training_provider_id: providers_without_published_courses.select(:id),
+    )
+    .where.not(
+      training_provider_id: previous_cycle_discarded_providers.select(:id),
+    )
+    .where.not(
+      accredited_provider_id: previous_cycle_discarded_providers.select(:id),
+    )
+  end
+
+  def rolled_over_partnerships
+    ProviderPartnership.includes(:training_provider, :accredited_provider).where(
+      "accredited_provider_id IN (:id) OR training_provider_id IN (:id)",
+      id: @target_cycle.providers.select(:id),
+    )
+  end
+
+  def partnerships_diff
+    previous_pairs = eligible_partnerships.map do |partnership|
+      [
+        partnership.training_provider.provider_code,
+        partnership.accredited_provider.provider_code,
+      ]
+    end
+
+    rolled_over_pairs = rolled_over_partnerships.map do |partnership|
+      [
+        partnership.training_provider.provider_code,
+        partnership.accredited_provider.provider_code,
+      ]
+    end
+
+    previous_pairs - rolled_over_pairs
```

Comment:
I created this method for my own sanity in the rails console. I am thinking to create a card where I added a tab for partnerships that wasn't rolled over. 
For now I decided to commit and push and might be useful for debug or investigating. Let me know if you wanna this method to not be committed.